dds/DCPS/transport/shmem/ShmemTransport.cpp

```diff
@@ -116,6 +115,7 @@ ShmemTransport::configure_i(TransportInst* config)
                     ACE_TEXT("no platform support for shared memory!\n")),
                    false);
 #else
+  ShmemInst* conf = dynamic_cast<ShmemInst*>(config);
```

Comment:
It fixes a compile warning I did get in one of our test builds when testing this change, updated the PR title to reflect that